### PR TITLE
COMMERCE-5446 Price list qualifiers query wrong filters

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java [hunk 1]
```diff
@@ -338,6 +338,82 @@ public void testRetrieveCorrectPriceListByHierarchy() throws Exception {
...
+				"applicable to the same account on another channel"
+		).given(
+			"A catalog with a price list on account and channel"
+		).when(
+			"The price list is discovered given a different channel"
+		).then(
+			"Only the catalog base price list is returned"
+		);
+
+		CommerceCatalog catalog =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java:354]:
```suggestion
		CommerceCatalog commerceCatalog =
```

File: modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java [hunk 2]
```diff
@@ -338,6 +338,82 @@ public void testRetrieveCorrectPriceListByHierarchy() throws Exception {
...
+			_commerceCatalogLocalService.addCommerceCatalog(
+				RandomTestUtil.randomString(), _commerceCurrency.getCode(),
+				LocaleUtil.US.getDisplayLanguage(), null, _serviceContext);
+
+		CommercePriceListTestUtil.addAccountAndChannelPriceList(
+			catalog.getGroupId(), _commerceAccount1.getCommerceAccountId(),
+			_commerceChannel1.getCommerceChannelId(), _TYPE);
+
+		CommercePriceList discoveredPriceList =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java:363]:
```suggestion
		CommercePriceList discoveredCommercePriceList=
```

File: modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java [hunk 3]
```diff
@@ -338,6 +338,82 @@ public void testRetrieveCorrectPriceListByHierarchy() throws Exception {
...
+				"not applicable to the same account on another channel"
+		).given(
+			"A catalog with a price list on account group and channel"
+		).when(
+			"The price list is discovered given a different channel"
+		).then(
+			"Only the catalog base price list is returned"
+		);
+
+		CommerceCatalog catalog =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-price-list-test/src/testIntegration/java/com/liferay/commerce/price/list/pricing/test/CommercePriceListHierarchyDiscoveryTest.java:390]:
```suggestion
		CommerceCatalog commerceCatalog =
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**